packages/shared/lib/services/sync/config/deploy.service.ts

```diff
@@ -75,6 +77,10 @@ export async function deploy(
     });
 
     const activityLogId = await createActivityLog(log);
+    const logCtx = await getOperationContext(
+        { id: String(activityLogId), operation: { type: 'deploy', action: 'custom' }, message: 'Deploying custom syncs' },
+        { account: { id: accountId }, environment: { id: environment_id } }
+    );
```

Comment:
you probably have tried but I have to ask since one of challenge of dealing with the codebase is the tangled dependencies: is it impossible to inject the `getOperationContext` or any logs high level function into `shared` so it doesn't have to required and loaded by any component using shared that doesn't also need `logs`?

Reply:
it's not impossible, but requires a lot more work and it's weird to pass just a function since it's a not singleton. However I could make a class LogContextGetter (or whatever name) to make it clear.

Since I'm stuck with the CLI, I might try